addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/WebCacheDeceptionScanRule.java

```diff
@@ -0,0 +1,182 @@
+    private static final Logger LOG = LogManager.getLogger(WebCacheDeceptionScanRule.class);
+
+    protected static final String[] TEST_EXTENSIONS = {
+        "css", "jpg", "js", "html", "gif", "png", "svg", "php", "txt", "pdf", "asp"
+    };
+
+    @Override
+    public int getId() {
+        return 40039;
+    }
+
+    @Override
+    public String getName() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "name");
+    }
+
+    @Override
+    public String getDescription() {
+        return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+    }
+
+    @Override
+    public void scan() {
+        if (initialTest()) {
+            HttpMessage authorisedMessage = getNewMsg();
+            try {
+                sendAndReceive(authorisedMessage);
+                String originalResponse = authorisedMessage.getResponseBody().toString();
+                URI uri = authorisedMessage.getRequestHeader().getURI();
+                ArrayList<String> extensions = new ArrayList<String>();
+                String path = uri.getPath();
+                // checks whether the page with appended path gets cached or not
+                for (String ext : TEST_EXTENSIONS) {
+                    String newPath = path + "/test." + ext;
+                    uri.setPath(newPath);
+                    authorisedMessage.getRequestHeader().setURI(uri);
+                    sendAndReceive(authorisedMessage);
+                    String cachedResponse = authorisedMessage.getResponseBody().toString();
+
+                    HttpMessage unauthorisedMessage = new HttpMessage(uri);
+                    unauthorisedMessage
+                            .getRequestHeader()
+                            .setMethod(authorisedMessage.getRequestHeader().getMethod());
+                    sendAndReceive(unauthorisedMessage);
+                    String unauthorisedCachedResponse =
+                            unauthorisedMessage.getResponseBody().toString();
+                    // similarity between authorised and unauthorised response is checked
+                    if (checkSimilarity(unauthorisedCachedResponse, cachedResponse)) {
+                        extensions.add(ext);
+                    }
+                }
+                uri.setPath(path);
+                authorisedMessage.getRequestHeader().setURI(uri);
+                sendAndReceive(authorisedMessage);
+                if (!extensions.isEmpty()) {
+                    String attack = "";
+                    for (String ext : extensions) {
+                        attack = attack + "/test." + ext + ",";
+                    }
+                    newAlert()
+                            .setConfidence(Alert.CONFIDENCE_MEDIUM)
+                            .setMessage(authorisedMessage)
+                            .setOtherInfo(
+                                    "Cached Authorised Response and Unauthorised Response are similar.")
```

Comment:
Sorry mi missed this earlier :( It seems you added the i18n string but then didn't use it.
```suggestion
                                    Constant.messages.getString("ascanalpha.webCacheDeception.otherinfo"))
```